design-documents/graph-ql/coverage/catalog/configurable-options-selection.graphqls

```diff
@@ -0,0 +1,14 @@
+type Query {
+    configurableOptionsSelectionMetadata(configurableProductSku: ID!, selectedConfigurableOptionValues: [ID!]): ConfigurableOptionsSelectionMetadata @doc(description: "Get metadata for the specified configurable options selection")
```

Comment:
Configurable options can't exist outside of a product, right?

Seems like this might make more sense as a field under `ConfigurableProduct`? Right now it seems like we're namespacing via naming convention instead of type.

One benefit to moving to the `ConfigurableProduct` type is that the client doesn't _need_ to know the `configurableProductSku` ahead of time - this covers the case where the UI wants to render in a single request with default selections, and doesn't know the `sku` before the request (UI fetches product using a `url_key` filter with `eq` matcher in this scenario).

```graphql
query ConfigurableOptionsAvailable($sku: ID!, $selectedOptions: [ID!]!) {
   products(filter: { sku: { eq: "foo" } }) {
	 items {
        ...on ConfigurableProduct {
           options_selections(selectedOptions: $selectedOptions) {
              # select fields
           }
        }
     }
   }
}
```

Reply:
I would say this depends on whether we want to support this functionality in category/search or not, discussed below https://github.com/magento/architecture/pull/394#discussion_r451720264

Reply:
Worth keeping in mind, you can get a link to a product page with pre-selected options.

For example, the client we discussed on the call yesterday includes selected option IDs in the URL for their product pages. If that's linked to someone, when they load the page the UI should be able to get the new selection set, without requiring a second round-trip (and client doesn't have sku from the URL itself, but does have `url_key`).

Reply:
There will be two calls in my understanding: 
1. urlResolver
2. Combined query: product details (using new query by ID/sku) + available selections query

URL will have to contain selections in compatible format, that is currently used in add to cart with single mutation.

Reply:
Commenting for posterity: this was resolved via the recent change made for an unrelated comment. 2 birds, 1 stone!

Reply:
For posterity: This was resolved via a change made to address a different comment. 2 birds, 1 stone 😄